src/libs/ajax.js

```diff
@@ -37,18 +30,15 @@ const withInstrumentation = wrappedFetch => (...args) => {
   )(ajaxOverridesStore.get())
 }
 
-const withCancellation = wrappedFetch => (...args) => {
-  return new Promise(async (resolve, reject) => {
-    try {
-      resolve(await wrappedFetch(...args))
-    } catch (error) {
-      if (error instanceof DOMException && error.name === 'AbortError') {
-        // no-op, this is from an aborted call
-      } else {
-        reject(error)
-      }
+const withCancellation = wrappedFetch => async (...args) => {
+  try {
+    return await wrappedFetch(...args)
+  } catch (error) {
+    if (error instanceof DOMException && error.name === 'AbortError') {
+      return Utils.abandonedPromise()
     }
```

Comment:
I like an explicit `else` here, it indicates to me that the things that happen in the conditional aren't just side-effects.